avm/res/container-service/managed-cluster/main.bicep

```
@@ -628,6 +638,10 @@ resource managedCluster 'Microsoft.ContainerService/managedClusters@2024-03-02-p
       keda: {
         enabled: kedaAddon
       }
+      verticalPodAutoscaler: {
+        addonAutoscaling: vpaAddonAutoscaling
+        enabled: vpaAddon
+      }
```

Perhaps this should be something like this: 
```suggestion
      verticalPodAutoscaler: {
        addonAutoscaling: (vpaAddon != false) ? vpaAddonAutoscaling : 'Disabled'
        enabled: vpaAddon
      }
```

---

Seems addonAutoscaling is in preview, so i have removed it.